app/controllers/line_items_controller.rb

```diff
@@ -1,7 +1,10 @@
 class LineItemsController < BaseController
   respond_to :json
 
-  def index
+  # Taken from Spree::Api::BaseController
+  rescue_from ActiveRecord::RecordNotFound, :with => :not_found
```

Comment:
As you wish, I'm generally used to make clear in the controller what each action need and how that affects the response. Basically: `not_found` implemented in a `Base` controller and `before_filter` in endpoint controller.